server/internal/worker/worker.go

```diff
@@ -0,0 +1,136 @@
+package worker
+
+import (
+	"sync"
+	"sync/atomic"
+)
+
+// A Job represents a unit of work to be done by a worker in the pool.
+type Job struct {
+	// job == nil is used as a sentinal value to instruct workers to quit
+	job    interface{}
+	commit func(result interface{})
+}
+
+// NewJob creates a new Job. job will be provided as the input to a worker's Do
+// method, and once the Do method returns, its return value will be passed as
+// the argument to commit.
+func NewJob(job interface{}, commit func(result interface{})) Job {
+	if job == nil {
+		panic("nil job")
+	}
+	return Job{job: job, commit: commit}
+}
+
+// An Idler performs a unit of background work when Idle is called.
```

Comment:
Yes, but nothing related to this. We create an idler to generate ECDSA random values [here](https://github.com/cloudflare/gokeyless/pull/175/files#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bR455).